src/plugProjectOgawaU/ogScreen.cpp

```diff
@@ -1,1880 +1,461 @@
-lfs      f2, lbl_8051D540@sda21(r2)
-fsub     f3, f4, f0
-lfs      f0, lbl_8051D51C@sda21(r2)
-fmuls    f1, f2, f1
-lfs      f2, lbl_8051D544@sda21(r2)
-fdiv     f3, f4, f3
-fmul     f1, f3, f1
-frsp     f1, f1
-fcmpo    cr0, f1, f0
-bge      lbl_80302730
-lfs      f0, lbl_8051D514@sda21(r2)
-lis      r3, sincosTable___5JMath@ha
-addi     r3, r3, sincosTable___5JMath@l
-fmuls    f0, f1, f0
-fctiwz   f0, f0
-stfd     f0, 8(r1)
-lwz      r0, 0xc(r1)
-rlwinm   r0, r0, 3, 0x12, 0x1c
-lfsx     f0, r3, r0
-fneg     f1, f0
-b        lbl_80302754
-
-lbl_80302730:
-lfs      f0, lbl_8051D518@sda21(r2)
-lis      r3, sincosTable___5JMath@ha
-addi     r3, r3, sincosTable___5JMath@l
-fmuls    f0, f1, f0
-fctiwz   f0, f0
-stfd     f0, 0x10(r1)
-lwz      r0, 0x14(r1)
-rlwinm   r0, r0, 3, 0x12, 0x1c
-lfsx     f1, r3, r0
-
-lbl_80302754:
-lfs      f0, lbl_8051D530@sda21(r2)
-fmadds   f1, f2, f1, f0
-
-lbl_8030275C:
-addi     r1, r1, 0x20
-blr
-	*/
-}
+	float ratio = p1 / p2;
 
-/*
- * --INFO--
- * Address:	80302764
- * Size:	0000B0
- */
-// void calcGlbCenter__Q22og6ScreenFP7J2DPaneP10Vector2<float>(void)
-void calcGlbCenter(J2DPane*, Vector2f*)
-{
-	/*
-	stwu     r1, -0x40(r1)
-	mflr     r0
-	li       r5, 0
-	stw      r0, 0x44(r1)
-	stw      r31, 0x3c(r1)
-	mr       r31, r4
-	stw      r30, 0x38(r1)
-	mr       r30, r3
-	addi     r3, r1, 0x14
-	mr       r4, r30
-	bl       getGlbVtx__7J2DPaneCFUc
-	lwz      r5, 0x14(r1)
-	mr       r4, r30
-	lwz      r6, 0x18(r1)
-	addi     r3, r1, 8
-	lwz      r0, 0x1c(r1)
-	stw      r5, 0x2c(r1)
-	li       r5, 3
-	stw      r6, 0x30(r1)
-	stw      r0, 0x34(r1)
-	bl       getGlbVtx__7J2DPaneCFUc
-	lwz      r0, 8(r1)
-	lwz      r3, 0xc(r1)
-	stw      r0, 0x20(r1)
-	lwz      r0, 0x10(r1)
-	stw      r3, 0x24(r1)
-	lfs      f3, 0x2c(r1)
-	lfs      f2, 0x20(r1)
-	lfs      f1, 0x30(r1)
-	lfs      f0, 0x24(r1)
-	fadds    f3, f3, f2
-	lfs      f2, lbl_8051D520@sda21(r2)
-	fadds    f0, f1, f0
-	stw      r0, 0x28(r1)
-	fmuls    f1, f3, f2
-	fmuls    f0, f0, f2
-	stfs     f1, 0(r31)
-	stfs     f0, 4(r31)
-	lwz      r31, 0x3c(r1)
-	lwz      r30, 0x38(r1)
-	lwz      r0, 0x44(r1)
-	mtlr     r0
-	addi     r1, r1, 0x40
-	blr
-	*/
-}
+	if (ratio < 0.0f) {
+		ratio = 0.0f;
+	}
+	if (ratio > 1.0f) {
+		ratio = 1.0f;
+	}
 
-/*
- * --INFO--
- * Address:	........
- * Size:	000154
- */
-void kakomiPane(J2DPane*)
-{
-	// UNUSED FUNCTION
+	float limit = 0.8f;
+
+	if (ratio < limit) {
+		return ratio;
+	}
+
+	float theta = ((1.0 / (1.0 - (double)limit)) * (double)(HALF_PI * (ratio - limit)));
+	float sin   = pikmin2_sinf(theta);
+	return (0.19999999f * sin) + 0.8f;
 }
 
 /*
  * --INFO--
- * Address:	........
- * Size:	000224
+ * Address:	80302764
+ * Size:	0000B0
  */
-void centeringPane(J2DPane*, bool, bool)
+void calcGlbCenter(J2DPane* pane, Vector2f* center)
 {
-	// UNUSED FUNCTION
+	Vector3f vec1 = pane->getGlbVtx(0);
+	Vector3f vec2 = pane->getGlbVtx(3);
+	center->x     = (vec1.x + vec2.x) / 2.f;
+	center->y     = (vec1.y + vec2.y) / 2.f;
```

Comment:
Nit: I'd prefer `2` over `2.f` here since we don't use this syntax anywhere else

Reply:
agreed

Reply:
yeah this was a copy-paste from encounter's solution - I'll admit I didn't investigate whether it was required or not for it to match. will double check and update accordingly